lib/streamlit/runtime/caching/cache_resource_api.py

```diff
@@ -473,7 +473,7 @@ def __init__(
         super().__init__()
         self.key = key
         self.display_name = display_name
-        self._mem_cache: TTLCache[str, MultiCacheResults] = TTLCache(
+        self._mem_cache: TimedCleanupCache[str, MultiCacheResults] = TimedCleanupCache(
```

Comment:
Hm, do we currently have examples of `@st.cache_resource` being the source of significant memory growth problems in deployed apps? Spinning up one of these periodic expiry tasks for `MemorySessionStorage` doesn't worry me too much since we know there will only be one of them, but for `@st.cache_resource` we could end up with quite a few of them. It's unlikely this ends up being a problem in practice since I doubt many apps have _that_ many functions that use the decorator, but we may want to hold off on replacing too many `TTLCache`s with `TimedCleanupCache`s to start.

Reply:
Nope, I did it on the hunch that avoiding a single big cache entry being retained after expiration would be much more memory than the total memory used by the small tasks this would introduce.